src/besselk.jl

```diff
@@ -431,3 +434,63 @@ end
 besselk_power_series_cutoff(nu, x::Float64) = x < 2.0 || nu > 1.6x - 1.0
 besselk_power_series_cutoff(nu, x::Float32) = x < 10.0f0 || nu > 1.65f0*x - 8.0f0
 
+
+"""
+  besselk_asymptoticexp(v, x, order)
+
+Computes the asymptotic expansion of K_ν w.r.t. argument. Accurate for large x, and faster than uniform asymptotic expansion for small to small-ish orders. The default order of the expansion in `Bessels.besselk` is 10.
+"""
+besselk_asexp_cutoff(nu, x::T) where T = (nu < 20.0) && (x > ASEXP_CUTOFF(T))
+
+# Compute the function for (_v, _v+1), where _v = v-floor(v), and then do
+# forward recursion. If the argument is really large, this should be faster than
+# the uniform asymptotic expansion and plenty accurate.
+function besselk_asymptoticexp(v, x::T, order) where T
+    _v  = v - floor(v)
+    (kv, kvp1) = _besselk_as_pair(_v, x, order)
+    v == _v && return kv
+    _v += one(v)
+    twodx = T(2)*inv(x)
+    while _v < v
+        (kv, kvp1) = (kvp1, muladd(kvp1, twodx*_v, kv))
+        _v += one(_v)
+    end
+    kvp1
+end
+
+# For now, no exponential improvement. It requires the exponential integral
+# function, which would either need to be lifted from SpecialFunctions.jl or
+# re-implemented. And with an order of, like, 10, this seems to be pretty
+# accurate and still faster than the uniform asymptotic expansion.
+function _besselk_as_pair(v, x::T, order) where{T}
```

Comment:
```suggestion
function _besselk_as_pair(v, x::T, order) where T
    fv = 4*v*v
    fvp1 = 4*(v+one(v))^2
    z = 8*x
    knu, knpu1 = one(T), one(T)
    ak_nu = fv - 1
    ak_nup1 = fvp1 - 1

    for i in 1:order
        term_v = ak_nu / z
        term_vp1 = ak_nup1 / z
        knu += term_v
        knpu1 += term_vp1

        a = muladd(2, i, one(T))^2
        z *= 8*x*(i + one(T))

        ak_nu *= fv - a
        ak_nup1 *= fvp1 - a
    end
    coef = SQRT_PID2(T)*exp(-x)/sqrt(x)
    return coef*knu, coef*knpu1
end
```

Reply:
These 2 divisions by `z` aren't combinable.

Reply:
This is almost 10% faster and a little bit more clear. 
```julia
# old
julia> @benchmark _besselk_as_pair(v, x, 60) setup=(v=rand(); x = rand()+50.0)
BenchmarkTools.Trial: 10000 samples with 957 evaluations.
 Range (min … max):  89.864 ns … 149.512 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     90.082 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   91.096 ns ±   3.844 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █▂      ▃▂                                                   ▁
  ██▇▇▇▇▇▆██▇▆▇▇▇▇▇▆▆▇▆▅▅▅▅▅▅▄▄▅▄▅▆▅▁▄▆▅▆▆▅▆▅▄▆▆▅▆▆▆▇▄▅▅▆▅▄▅▄▆ █
  89.9 ns       Histogram: log(frequency) by time       111 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.


# new
julia> @benchmark _besselk_as_pair(v, x, 60) setup=(v=rand(); x = rand()+50.0)
BenchmarkTools.Trial: 10000 samples with 967 evaluations.
 Range (min … max):  81.135 ns … 126.897 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     81.352 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   81.575 ns ±   1.414 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

   ▆█▇▄▄                                ▁▃                     ▂
  ███████▇▆▄▁▃▅▃▄▄▁▅▃▃▅▅▄▄▅▅▃▃▄▄▁▁▁▁▄▃▄▇███▅▅▃▅▄▃▁▄▃▃▃▁▁▁▁▃▄▅▄ █
  81.1 ns       Histogram: log(frequency) by time      85.7 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```

Oscar might have some better suggestions here.

Reply:
Yes - and this is almost vectorizing away completely anyway. It takes 76 ns to compute just a single order so getting both `nu` and `nu+1` here are almost free.